packages/server/src/workers/subscription.ts

```diff
@@ -263,34 +263,38 @@ export async function execSubscriptionJob(job: Job<SubscriptionJobData>): Promis
     // If the subscription has been disabled, then stop processing it.
     return;
   }
+  const deletedInteraction = isValidDeleteInteraction(subscription);
+
+  if (!deletedInteraction) {
+    let currentVersion;
+    try {
+      currentVersion = await systemRepo.readResource(resourceType, id);
+    } catch (err) {
+      const outcome = normalizeOperationOutcome(err);
+      if (isGone(outcome)) {
+        // If the resource was deleted, then stop processing it.
+        return;
+      }
+      // Otherwise re-throw
+      throw err;
+    }
 
-  let currentVersion;
-  try {
-    currentVersion = await systemRepo.readResource(resourceType, id);
-  } catch (err) {
-    const outcome = normalizeOperationOutcome(err);
-    if (isGone(outcome)) {
-      // If the resource was deleted, then stop processing it.
+    if (job.attemptsMade > 0 && currentVersion.meta?.versionId !== versionId) {
+      // If this is a retry and the resource is not the current version, then stop processing it.
       return;
     }
-    // Otherwise re-throw
-    throw err;
-  }
-
-  if (job.attemptsMade > 0 && currentVersion.meta?.versionId !== versionId) {
-    // If this is a retry and the resource is not the current version, then stop processing it.
-    return;
   }
 
   try {
-    const resourceVersion = await systemRepo.readVersion(resourceType, id, versionId);
-
+    const resourceVersion = deletedInteraction
+      ? { resourceType, id }
```

Comment:
This morning we discussed sending empty string - this will send content.  

Reading through... that might make this more complicated, because both `execBot` and `sendRestHook` require a resource...

This is spiralling.  Let's discuss offline.